spmd/compiler/fusion.py

```diff
@@ -736,9 +736,9 @@ def run_overlap_communication(gm: fx.GraphModule) -> None:
     for i, item in enumerate(fe_list[1:]):  # type: ignore
```

Comment:
I think adding `i = -1` before this line is more accurate.

Reply:
makes sense. We tend not to refer to loop vars outside the loop and hence assumed we were debugging something else.

Also make more sense for us to assert that we should have optimized "something" at this point so we should not asserting on i > -1?